src/authentication/views.py

```diff
@@ -0,0 +1,104 @@
+]
+
+
+class FormViewMixin:
+    """A class encapsulating features common to all of the views related to
+    authentication actions using forms."""
+
+    # The following attributes are intentionally set to None in order to enforce
+    # specifying them (or respective hooks) in child classes of the base  views. If
+    # this is not done, `ImproperlyConfigured` is raised.
+    success_url = None
+    template_name = None
+
+
+class LoginView(FormViewMixin, LoginView):
+    """A base class-based view to handle logging in."""
+
+    form_class = AuthenticationForm
+
+    def form_valid(self, form):
+        # Override the base method to prevent inactive users from logging in.
+        user = form.get_user()
+        if not user.is_active:
+            return self.form_invalid(form)
+        return super().form_valid(form)
+
+
+class LogoutView(LogoutView):
+    """A base class-based view to handle logging out."""
+
+
+class PasswordChangeView(FormViewMixin, PasswordChangeView):
+    """A base class-based view to handle password change."""
+
+    form_class = PasswordChangeForm
+
+
+class PasswordResetRequestView(FormViewMixin, PasswordResetView):
+    """A base class-based view to handle password reset request."""
+
+    email_template_name = "authentication/password_reset_request_email.html"
+    form_class = PasswordResetRequestForm
+    subject_template_name = "authentication/password_reset_request_email_subject.txt"
+
+    def dispatch(self, request, *args, **kwargs):
+        # Override the base method to update the context available in the password
+        # reset request email.
+        extra_email_context = self.extra_email_context or {}
+        extra_email_context.setdefault(
+            "password_reset_timeout", settings.PASSWORD_RESET_TIMEOUT // 60
+        )
+        self.extra_email_context = extra_email_context
+        return super().dispatch(request, *args, **kwargs)
+
+
+class PasswordResetConfirmView(FormViewMixin, PasswordResetConfirmView):
+    """A base class-based view to handle password reset confirmation."""
+
+    form_class = PasswordResetConfirmForm
+    invalid_token_url = None
+    success_url = reverse_lazy(settings.LOGIN_URL)
+
+    def render_to_response(self, context, **response_kwargs):
+        # Override the base method to redirect to `invalid_token_url`. Note that a
+        # standard render response is returned; thus, this method is triggered only as
+        # a result of a request with an invalid (or expired) token.
+        if not self.validlink:
+            return redirect(self.invalid_token_url)
+        return super().render_to_response(context, **response_kwargs)
+
+    def get_invalid_token_url(self):
+        """Return the URL to which the user should be redirected if the
+        password reset token is invalid or expired."""
+        if not self.invalid_token_url:
+            raise ImproperlyConfigured(
+                "No URL to redirect to. Provide a 'invalid_token_url'."
```

Comment:
Update the error message:
```suggestion
                "There is no URL to redirect to. Provide a value for the 'invalid_token_url' attribute."
```